### PR TITLE
[MSVC] added possible to compile at Microsoft Visual C compiler.

### PR DESCRIPTION
**Pull request by TheVice**
Hello.

In this pull request resolving compile errors at C compiler from Microsoft, described [here](https://github.com/BLAKE3-team/BLAKE3/issues/47).
If you are also interesting of linking, please also read [this issue](https://github.com/BLAKE3-team/BLAKE3/issues/16) - linking at MSVC environment have same issue like for the gcc.

Thank you.

### CODE REVIEW

File: c/blake3_neon.c [hunk 1]
```diff
@@ -1,5 +1,5 @@
 #include "blake3_impl.h"
-
+#if !defined(_WIN32)
```
Comment (oconnor663) [inline c/blake3_neon.c]: Why this change? Anything not explicitly targeting ARM NEON should be skipping this file entirely.

File: c/main.c [hunk 2]
```diff
@@ -4,7 +4,9 @@
 #include <stdint.h>
 #include <stdio.h>
 #include <stdlib.h>
+#if !defined(_WIN32)
 #include <unistd.h>
```
Comment (oconnor663) [inline c/main.c]: Maybe we can remove this header entirely? It doesn't seem to affect the build when I try it.

At the same time, if we're adding Windows support here, we should probably figure out a way to add Windows CI testing in `.github/workflows/ci.yml`.

File: c/blake3_neon.c (hunk 1 above)
Comment (TheVice) [reply c/blake3_neon.c]: Guarding content of blake3_neon.c is recall/same approach used at request about ['using C code in C++'](https://github.com/BLAKE3-team/BLAKE3/pull/40). But in this case we have choose:
- even do not include this source file, through CMakeLists.txt, Visual Studio project, what ever other thing that able to build source files;
- or just guard file via macros - so no mater how we create project in this case.
It is no problem to not include this macros, because comparing to "external 'C'" construction such approach not so popular.

In case of "external 'C'" another approach will be that C++ side include C file inside proposed construction. 
```
extern "C" {
#include "blake3.h"
};
```

File: c/main.c (hunk 2 above)
Comment (TheVice) [reply c/main.c]: Yes, for current content of main.c, including of this file can be removed. Probably it was left some experimentations that even not was committed.

Agree, adding of Windows CI testing environment would be right step to protect all changes proposed here and prevent regression changing that broke compile able at MSVC.

**Comment by oconnor663** [comment]
One of the commits in this PR has a malformed commit message. A blank line between the summary line and the rest of the commit message is important, to avoid screwing up the formatting of `git log --oneline`. Could you please rebase and squash for commits, with a cleaned up message?

**Comment by TheVice** [comment]
Sure.

Branch was re-based, commit are squashed into single one.

First one was created at TortoiseGit, second one at Git-Cola, where exists one-line control at the GUI.

**Comment by oconnor663** [comment]
Thanks! Merging.
